src/conta/Conta.test.js

```diff
@@ -1,19 +1,29 @@
 import React from 'react';
-import { render, fireEvent, screen } from '@testing-library/react';
-import ContaService from './ContaService';
+import axios from 'axios';
+import { render, fireEvent, screen, waitForElementToBeRemoved } from '@testing-library/react';
 import Conta from './Conta';
 
+jest.mock('axios');
+
+afterEach(() => {
+  axios.get.mockClear();
+})
+
+beforeEach(() => {
+  axios.get.mockResolvedValue({data: {saldo: 1234.57}});
+});
+
 describe('Componente da Conta', () => {
-  it('renderiza Conta', () => {
+  it('renderiza Conta', async () => {
     render(<Conta />);
+    await waitForElementToBeRemoved(() => screen.getByText('carregando'));
```

Comment:
Se esse elemento de carregando nunca sair da tela. Seu teste trava aqui? O Jest tem um timeout por padrão? Se for isso que vc quer testar o nome do teste precisa ser mais especifico

Reply:
Sim, o jest tem timeout e vai falhar. A questão é que ele serve mais para esperar o componente terminar de renderizar por inteiro, pois depois que a Promise do axios é resolvida, o componente muda de estado. Caso não esperar isso, ele da um warning avisando que o componente foi alterado depois que o teste terminou.
Existem algumas soluções, esperar a promise ou esperar o componente ficar no estado final. Eu prefiro validar olhando o componente, assim já garante algumas coisas no componente.
O que acha?

Reply:
Faz sentido a validação das duas maneiras. EM um caso vc ta mais próximo dos testes de caixa-branca no outro dos testes de caixa-preta. A maneira que vc prefere é mais o segundo e como o resto do projeto ta indo pra essa linha tudo bem